### PR TITLE
Add info about cosign

### PR DESCRIPTION
**Pull request by Itxaka**
Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

File: content/en/docs/Creating derivatives/creating_bootable_images.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ The image needs to ship:
 - grub (required)
 - dracut (optional, kernel and initrd can be consumed from the cOS repositories)
 - microcode (optional, not required in order to boot, but recomended)
+- cosign and luet-cosign packages (optional, required if you want to verify the images installed by luet)
```
Comment (mudler) [inline content/en/docs/Creating derivatives/creating_bootable_images.md:28]: small nit.. maybe add a link here to the docs below?

File: content/en/docs/Creating derivatives/creating_bootable_images.md [hunk 2]
```diff
@@ -55,6 +62,8 @@ In the example above, the cos-toolkit parts that are **required** are pulled in
 {{<package package="system/cloud-config" >}} is optional, but provides `cOS` defaults setting, like default user/password and so on. If you are not installing it directly, an equivalent cloud-config has to be provided in order to properly boot and run a system, see [oem configuration](../../customizing/oem_configuration).
 {{% /alert %}}
 
+The {{<package package="toolchain/cosign" >}} and {{<package package="toolchain/luet-cosign" >}} are optional packages that would install cosign and luet-cosign in order to verify the packages installed by luet in the next step. For more info, check the [cosign](../../getting-started/cosign) page.
```
Comment (mudler) [inline content/en/docs/Creating derivatives/creating_bootable_images.md:65]: Maybe turn this into small paragraph about it ? feels like this information would get a bit lost

File: content/en/docs/Getting started/cosign.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+
+---
+title: "Cosign"
+linkTitle: "Cosign"
```
Comment (mudler) [inline content/en/docs/Getting started/cosign.md:4]: :+1: great!

File: content/en/docs/Getting started/cosign.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+weight: 2
+date: 2017-01-05
```
Comment (mudler) [inline content/en/docs/Getting started/cosign.md:6]:
```suggestion
date: 2020-11-02
```

File: content/en/docs/Getting started/cosign.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+description: >
+  How we use cosign in cos-toolkit
+---
+
+Cosign is a project that signs and verifies containers and stores the signatures on OCI registries.
```
Comment (mudler) [inline content/en/docs/Getting started/cosign.md:11]:
```suggestion
[Cosign](https://github.com/sigstore/cosign) is a project that signs and verifies containers and stores the signatures on OCI registries.
```

**Review by mudler — APPROVED** [review]
Just small nit, otherwise looks good!

File: content/en/docs/Creating derivatives/creating_bootable_images.md (hunk 2 above)
Comment (Itxaka) [reply content/en/docs/Creating derivatives/creating_bootable_images.md:65]: well, its optional rigth? So I wanted to have it like the cloud-config package, but I can expand it a little bit

File: content/en/docs/Creating derivatives/creating_bootable_images.md [hunk 6]
```diff
@@ -38,10 +39,16 @@ FROM quay.io/luet/base:$LUET_VERSION AS luet
 FROM opensuse/leap:15.3 # or Fedora, Ubuntu
 ARG ARCH=amd64
 ENV ARCH=${ARCH}
+ENV COSIGN_EXPERIMENTAL=1 # keyless verify
+ENV COSIGN_REPOSITORY=raccos/releases-green  # repo with the signatures
 RUN zypper in -y ... # apt-get, dnf...
 
+# Here we install cosign and luet-cosign so we can verify that the images installed have been signed
+RUN luet install -y toolchain/cosign toolchain/luet-cosign
```
Comment (mudler) [inline content/en/docs/Creating derivatives/creating_bootable_images.md:47]: Something that "puzzles" me is that in this way, when we get cosign itself and the plugin, those are not verified (via signatures, etc). So we are with the chicken and egg problem, as our first bits that we install are not verified..

Shall we maybe create a signed container where we take this tools from instead?

Comment (Itxaka) [reply content/en/docs/Creating derivatives/creating_bootable_images.md:47]: how would we verify that container?

Comment (Itxaka) [reply content/en/docs/Creating derivatives/creating_bootable_images.md:47]: I guess what you propose is to then we can do:

```
COPY --from=cosign /usr/bin/cosign /usr/bin/cosign
COPY --from=cosign /usr/bin/luet-cosign /usr/bin/luet-cosign
```

???

File: content/en/docs/Creating derivatives/creating_bootable_images.md (hunk 2 above)
Comment (Itxaka) [reply content/en/docs/Creating derivatives/creating_bootable_images.md]: updated!

File: content/en/docs/Creating derivatives/creating_bootable_images.md (hunk 6 above)
Comment (mudler) [reply content/en/docs/Creating derivatives/creating_bootable_images.md:47]: yup, In that way if someone is _paranoic_ enough can just verify that container where it is taking the binary from

Comment (Itxaka) [reply content/en/docs/Creating derivatives/creating_bootable_images.md:47]: In that case, can you just get cosign manually and verify manually beforehand?

```
# itxaka @ zeus in ~/projects/cos-toolkit-docs on git:cosign o [11:07:00] C:1
$ export COSIGN_EXPERIMENTAL=1                                                  

# itxaka @ zeus in ~/projects/cos-toolkit-docs on git:cosign o [11:10:03] 
$ export COSIGN_REPOSITORY=raccos/releases-green

# itxaka @ zeus in ~/projects/cos-toolkit-docs on git:cosign o [11:10:10] 
$ cosign verify quay.io/costoolkit/releases-green:luet-cosign-toolchain-0.0.6-1               
No TUF root installed, using embedded CA certificate.

No TUF root installed, using embedded rekor key

Verification for quay.io/costoolkit/releases-green:luet-cosign-toolchain-0.0.6-1 --
The following checks were performed on each of these signatures:
  - The cosign claims were validated
  - Existence of the claims in the transparency log was verified offline
  - Any certificates were verified against the Fulcio roots.

[{"critical":{"identity":{"docker-reference":"quay.io/costoolkit/releases-green"},"image":{"docker-manifest-digest":"sha256:9377596ed7f98c084a86a96f0536ddbfacf7a3e90567e8b577745b602d4d6d3d"},"type":"cosign container image signature"},"optional":{"Bundle":{"SignedEntryTimestamp":"MEYCIQDSN/rJG/6+EsaaYrj8wpXP5ua4ecnRQZPv1sczWOXaewIhAMpx9k7IX8N0oEZXDAg657u48JjSCHALfmRiFHHX7nbw","Payload":{"body":"eyJhcGlWZXJzaW9uIjoiMC4wLjEiLCJraW5kIjoicmVrb3JkIiwic3BlYyI6eyJkYXRhIjp7Imhhc2giOnsiYWxnb3JpdGhtIjoic2hhMjU2IiwidmFsdWUiOiI4ZGRhMmNlMzYyOGRjM2FlMmIxM2RmMDhiYjc3NTM2NjVhNzcwZDMyMGYyMjE1MzZkOTlhM2I5ODZjMWNkYzg1In19LCJzaWduYXR1cmUiOnsiY29udGVudCI6Ik1FVUNJUUMrT0Y4ZUE2NnRkRWlnRUMyM29kZ21kaTgvNisxRmdaVHE2VHk1WGs1WXVRSWdQRFhQZ2JRYUpZYlJQQUkraVZQTzVNa3QydGdGMjVXK0ozdnpUdHArdjRZPSIsImZvcm1hdCI6Ing1MDkiLCJwdWJsaWNLZXkiOnsiY29udGVudCI6IkxTMHRMUzFDUlVkSlRpQkRSVkpVU1VaSlEwRlVSUzB0TFMwdENrMUpTVVJGVkVORFFYQmxaMEYzU1VKQlowbFZRVkEwZUhaYVVXNVJSMFl6SzFaMFEySlVSbXhKVWtWTVRtMXpkME5uV1VsTGIxcEplbW93UlVGM1RYY0tTMnBGVmsxQ1RVZEJNVlZGUTJoTlRXTXliRzVqTTFKMlkyMVZkVnBIVmpKTlVrVjNSSGRaUkZaUlVVUkZkMmg2WVZka2VtUkhPWGxhVkVGbFJuY3dlUXBOVkVWNFRVUkpkMDE2VlhkTlJFSmhSbmN3ZVUxVVJYaE5SRWwzVGtSQk5VNVViR0ZOUVVGM1YxUkJWRUpuWTNGb2EycFBVRkZKUWtKblozRm9hMnBQQ2xCUlRVSkNkMDVEUVVGUlVreDBUREJhT0V0SmFGcFNXRTh2VldvNWNITkdNeXMyWWs5VFIwZ3dNakpZT1ZwMlZXNWtabFZZZDB0cmVVVkNVSEExVkdnS2FWWjFXRWRDUzNNNVRFZDZPRXhuZVhwSGNUSjVaV3g2UW1KdFMyVXpVWGx2TkVsQ2QzcERRMEZpT0hkRVoxbEVWbEl3VUVGUlNDOUNRVkZFUVdkbFFRcE5RazFIUVRGVlpFcFJVVTFOUVc5SFEwTnpSMEZSVlVaQ2QwMUVUVUYzUjBFeFZXUkZkMFZDTDNkUlEwMUJRWGRJVVZsRVZsSXdUMEpDV1VWR1J6ZEtDa2RCV2pZeVNISkRUSGRqTWt3MGQyUXpaakZCV2tObmRrMUNPRWRCTVZWa1NYZFJXVTFDWVVGR1RXcEdTRkZDUW0xcFVYQk5iRVZyTm5jeWRWTjFNVXNLUW5SUWMwMUpSMDVDWjJkeVFtZEZSa0pSWTBKQlVWTkNaMFJDSzAxSWQwZERRM05IUVZGVlJrSjZRVU5vYmtKdlpFaFNkMDlwT0haalNFcHdaRzFHTUFwYVYwNW9URmRPZG1KdVVteGlibEYwVG1wQmVscHRWVE5hVkdOMFRVUkJkMDFETUhsTmFra3pURmRLYlU1NlZYUmFhbEp0VGxkVk5FMUhVWGxQVkZVd0NreHVUakJpTTBwb1dqSlZkVm95T1haYU1uaHNXVmhDY0dONU5XcGlNakIyV1RKRmVrNXRSWGhhVkdzeVRXcFJlVmxxYkcxWk1rbDRUa1JaZGxreVJYVUtXVE5LTUUxSU9FZEJNVlZrUlZGRlFpOTNVakZOU0U5SFkxZG9NR1JJUW5wUGFUaDJXakpzTUdGSVZtbE1iVTUyWWxNNWVWbFhOV3BoUjFaNVRGaE9hQXBpYlZKcFlqTm5kbGt3T1ZSTVdGSjJZako0Y21GWVVYWk1iV1J3WkVkb01WbHBPVE5pTTBweVdtMTRkbVF6VFhaWmJsWndZa2RSZEdKWFJucGtSMVo1Q2t4WFpIbGFWMVoxVEZobk5FNXNPREpPUXpVMVdWY3hjMUZJU214YWJrMTJZVWRXYUZwSVRYWmlWMFo2WkVkV2VVMUVhMGREYVhOSFFWRlJRbWMzT0hjS1FWRkZSVXN5YURCa1NFSjZUMms0ZG1SSE9YSmFWelIxV1ZkT01HRlhPWFZqZVRWdVlWaFNiMlJYU2pGak1sWjVXVEk1ZFdSSFZuVmtRelZxWWpJd2R3cERaMWxKUzI5YVNYcHFNRVZCZDAxRVlVRkJkMXBSU1hoQlNXSnJWRWxrYmpaNmFuZHpRMmhOYW5jeVFURjNOSHB3YkVkNVpHTnRZVVo0TXpOcGFXUmFDakpDWkN0UkwxaGthbmxxV0RaSk5GWldTMU15Y2xCV1lsVkJTWGRPYkhnMmEzaDNVbFl3UjBobGIwcDZVbmd6YURseWRFcDZjR0ZwT1ROR2EzRmFUVUVLWjI5VE5YcHpkRlZwUmxsd1JXRmFLMjVrUkdWaVRVdFpSVWRhT1FvdExTMHRMVVZPUkNCRFJWSlVTVVpKUTBGVVJTMHRMUzB0Q2c9PSJ9fX19","integratedTime":1635825001,"logIndex":816654,"logID":"c0d23d6ad406973f9559f3ba2d1ca01f84147d8ffc5b8445c224f98b9591801d"}},"Subject":"https://github.com/rancher-sandbox/cOS-toolkit/.github/workflows/build-master-green-x86_64.yaml@refs/heads/master"}}]

```

Comment (Itxaka) [reply content/en/docs/Creating derivatives/creating_bootable_images.md:47]: should probably add that to the docs :D

Comment (Itxaka) [reply content/en/docs/Creating derivatives/creating_bootable_images.md:47]: updated!

File: content/en/docs/Creating derivatives/creating_bootable_images.md [hunk 7]
```diff
@@ -55,6 +62,29 @@ In the example above, the cos-toolkit parts that are **required** are pulled in
...
+#### Using cosign in your derivative
+
+The {{<package package="meta/cos-verify" >}} is a meta package that will pull {{<package package="toolchain/cosign" >}} and {{<package package="toolchain/luet-cosign" >}} .
+
+{{<package package="toolchain/cosign" >}} and {{<package package="toolchain/luet-cosign" >}} are optional packages that would install cosign and luet-cosign in order to verify the packages installed by luet.
+
+You can use cosign to both verify that packages coming from cos-toolkit are verified and sign your own derivative artifacts
+
+{{% alert title="Note" %}}
+If you want to manually verify cosign and luet-cosign packages before installing them with luet, you can do so by:
```
Comment (mudler) [inline content/en/docs/Creating derivatives/creating_bootable_images.md:74]: Awesome! :partying_face:

**Review by mudler — APPROVED** [review]

**Pull request merged**